lib/checkstl.cpp

```diff
@@ -1654,6 +1659,8 @@ void CheckStl::missingComparison()
                     if (!tok3)
                         break;
                 }
+                else if (Token::Match(tok3, "%varid% + 1 )| !=|==", iteratorId))
```

Comment:
I think you follow the coding style well.. and that is in general a good idea. But this is an old check. I suggest that you try to use the syntax tree more. This check code should be modernized.